packages/client/lib/sync/fetcher/headerfetcher.ts

```diff
@@ -28,13 +33,17 @@ export class HeaderFetcher extends BlockFetcher {
    * Requests block headers for the given task
    * @param job
    */
-  async request(job: any) {
+  async request(job: Job<JobTask, BlockHeaderResult>) {
     const { task, peer } = job
-    if (this.flow.maxRequestCount(peer, 'GetBlockHeaders') < this.maxPerRequest) {
+    if (this.flow.maxRequestCount(peer!, 'GetBlockHeaders') < this.maxPerRequest) {
       // we reached our request limit. try with a different peer.
-      return false
+      return undefined
+    }
+    const response = await (peer!.les as LesProtocolMethods).getBlockHeaders({ block: task.first, max: task.count })
+    if (response) {
+      return response.headers
     }
-    return (peer.les as LesProtocolMethods).getBlockHeaders({ block: task.first, max: task.count })
+    return undefined
```

Comment:
I think here you can just do:

```return response?.headers```

or if `response` not an optional value:

```return response.headers ?? undefined```

Reply:
Don't know why I wrote this since I can just return the result 😅 . Fixed.